### PR TITLE
Allow course admins to delete (almost) unused courses

### LINKED ISSUE #1683: Add option to remove courses

### PR DESCRIPTION
**Pull request by niknetniko**
This pull request adds the possibility to delete courses:

- I've allowed course admins to delete their course. Is this what we want?
- Courses with more than 20 submissions cannot be deleted.
- A card with delete button is added to the course edit page.

Deletable course:
image

Undeletable course:
image

Closes #1683.

### CODE REVIEW

File: app/policies/course_policy.rb [hunk 1]
```diff
@@ -48,7 +50,7 @@ def update?
   end
 
   def destroy?
-    user&.zeus?
+    course_admin? && record.submissions.count <= MAX_SUBMISSIONS_FOR_DESTROY
```
Comment (charvp) [inline app/policies/course_policy.rb:53]: I think we discussed (somewhere) internally, that the submission limit should not count for zeus.

File: config/locales/views/courses/en.yml [hunk 2]
```diff
@@ -65,6 +65,15 @@ en:
       series_reordering_help_html: "You can reorder the series by dragging the icons at the left of the table. For students, a <b>reverse chronological</b> order is probably the easiest. This way, the most recent series is at the top, and the students don't have to scroll down each time."
       update: "Update"
       series_reordering_moved_help_html: 'Looking to reorder the series in this course? This functionality has moved to a <a href="%{course_url}">separate page</a>.'
+      delete:
+        title: "Delete course"
+        sure: "Deleting a course has following consequences."
```
Comment (charvp) [inline config/locales/views/courses/en.yml:70]:
```suggestion
        sure: "Deleting a course has the following consequences:"
```

File: config/locales/views/courses/nl.yml [hunk 3]
```diff
@@ -74,6 +74,15 @@ nl:
       series_reordering_help_html: "Je kan de reeksen herordenen door ze te verslepen met de icoontjes aan de linkerkant van de tabel. Voor studenten is een <b>omgekeerd chronologische</b> volgorde waarschijnlijk het gemakkelijkste. Hierbij staan recente reeksen bovenaan de pagina en hoeven ze niet telkens naar onder te scrollen."
       update: "Aanpassen"
       series_reordering_moved_help_html: "Wou je de reeksen in dit vak herordenen? Deze functionaliteit is verplaatst naar een <a href=\"%{course_url}\">aparte pagina</a>."
+      delete:
+        title: "Cursus verwijderen"
+        sure: "Het verwijderen van een cursus heeft volgende gevolgen:"
```
Comment (charvp) [inline config/locales/views/courses/nl.yml:79]:
```suggestion
        sure: "Het verwijderen van een cursus heeft de volgende gevolgen:"
```

**Comment by niknetniko** [comment]
@bmesuere it is now like this
image

File: config/locales/views/courses/nl.yml [hunk 4]
```diff
@@ -74,6 +74,16 @@ nl:
...
       update: "Aanpassen"
       series_reordering_moved_help_html: "Wou je de reeksen in dit vak herordenen? Deze functionaliteit is verplaatst naar een <a href=\"%{course_url}\">aparte pagina</a>."
+      delete:
+        title: "Cursus verwijderen"
+        consequences: "Het verwijderen van een cursus heeft de volgende gevolgen:"
+        submissions: "Oplossingen blijven bewaard, maar zijn niet meer gekoppeld aan een cursus."
+        series: "Reeksen gaan onherroepelijk verloren."
+        button: "Verwijderen"
+        impossible:
+          one: "Een cursus met meer dan %{count} oplossing kan niet verwijderd worden."
```
Comment (bmesuere) [inline config/locales/views/courses/nl.yml:84]:
```suggestion
          one: "Je kan een cursus met meer dan %{count} oplossing niet zelf verwijderen."
```

File: config/locales/views/courses/nl.yml [hunk 5]
```diff
@@ -74,6 +74,16 @@ nl:
...
+          other: "Een cursus met meer dan %{count} oplossingen kan niet verwijderd worden."
```
Comment (bmesuere) [inline config/locales/views/courses/nl.yml:85]:
```suggestion
          other: "Je kan een cursus met meer dan %{count} oplossingen niet zelf verwijderen."
```

File: config/locales/views/courses/en.yml [hunk 6]
```diff
@@ -65,6 +65,16 @@ en:
...
       update: "Update"
       series_reordering_moved_help_html: 'Looking to reorder the series in this course? This functionality has moved to a <a href="%{course_url}">separate page</a>.'
+      delete:
+        title: "Delete course"
+        consequences: "Deleting a course has the following consequences:"
+        submissions: "Submissions are kept, but are no longer linked to a course."
+        series: "Series are irrevocably lost."
+        button: "Delete"
+        impossible:
+          one: "A course with more than %{count} submission cannot be removed."
```
Comment (bmesuere) [inline config/locales/views/courses/en.yml:75]:
```suggestion
          one: "You can't remove a course with more than %{count} submission yourself."
```

File: config/locales/views/courses/en.yml [hunk 7]
```diff
@@ -65,6 +65,16 @@ en:
...
+          other: "A course with more than %{count} submissions cannot be removed."
```
Comment (bmesuere) [inline config/locales/views/courses/en.yml:76]:
```suggestion
          other: "You can't remove a course with more than %{count} submissions yourself."
```

**Pull request merged**